packages/components/organisms/f-checkout/src/components/ErrorDialog.vue

```diff
@@ -30,39 +34,52 @@ import MegaModal from '@justeat/f-mega-modal';
 import '@justeat/f-mega-modal/dist/f-mega-modal.css';
 import FButton from '@justeat/f-button';
 import '@justeat/f-button/dist/f-button.css';
-
-import EventNames from '../event-names';
+import { mapActions, mapState } from 'vuex';
+import {
+    VUEX_CHECKOUT_MODULE
+} from '../constants';
 
 export default {
     components: {
-        MegaModal,
-        FButton
+        FButton,
+        MegaModal
     },
 
-    props: {
-        isOpen: {
-            type: Boolean,
-            default: false
+    computed: {
+        ...mapState(VUEX_CHECKOUT_MODULE, [
+            'message',
+            'restaurant',
+            'serviceType'
+        ]),
+
+        errorCode () {
+            return this.message && this.message.code;
         },
-        errorCode: {
-            type: String,
-            default: ''
+
+        isOpen () {
+            return this.message && this.message.shouldShowInDialog;
```

Comment:
I might be wrong, but this looks a bit weird.
What if you've got a message and it should be shown in a dialog but it hasn't been shown yet?

Reply:
This has been removed